src/pending.js

```diff
@@ -0,0 +1,128 @@
+var _ = require('lodash');
+var debug = require('debug')('stats-collector:pending');
+var taskcluster = require('taskcluster-client');
+
+const FLUSH_PERIOD = 30; // seconds
+
+// check for pending tasks that aren't really pending with this frequency; should
+// be smaller than MIN_CHECK_AGE
+const CHECK_INTERVAL = 120; // seconds
+
+// only check tasks pending at least this long; this should be lower than any alert
+// thresholds for pending tasks.
+const MIN_CHECK_AGE = 300; // seconds
+
+// calculate the taskKey and scheduled date with the earliest scheduled time for
+// the given workerType; if there are none, returns null and the current time
+var earliest = (state, workerType, now) => {
+  return _.reduce(state.pendingTasks[workerType], (res, scheduled, taskKey) => {
```

Comment:
Using an array means operating on an array that is the length of the total pending -- so, spiking at exactly the time we want this service to give accurate information, and operating on that array using O(n) operations.  For example, to update task information using "normal" lodash methods, I'd use _.reject to remove the existing entry from the list, then add a new one, or something like that.

Some brief testing shows that `{}`'s performance is close enough to that of a hash table (property access is certainly sublinear in number of properties), and using this characteristic makes the `update` operation somewhere around O(log n), which I think is going to be key during high-load situations.

You're right that `_.minBy` doesn't operate on objects.  I initially wrote this based on `_.orderBy`, but it seemed like overkill since I only needed the first item of the sorted list.